src/test/scala/edu/ie3/simona/model/participant/BMModelSpec.scala

```diff
@@ -0,0 +1,338 @@
+        "MockNode",
+        isCostControlled = true,
+        EUR(bmType.getOpex.getValue.doubleValue()),
+        EuroPerKilowatthour(feedInTariff),
+        0.05,
+      )
+
+      val pElCalc = bmModel.calculateElOutput(usage, eff)
+      pElCalc.value should be(Kilowatts(pElSol).value +- 1e-4)
+    }
+  }
+
+  "apply load gradient correctly" in {
+    val bmModel = buildBmModel()
+
+    val testCases = Table(
+      ("Last Power", "PEl", "PEl Sol"),
+      (
+        Kilowatts(-100d), // Last Power
+        Kilowatts(-120d), // PEl
+        Kilowatts(-109.5d), // PEl Solution
+      ), // increase of power, more than load gradient allows
+      (
+        Kilowatts(-50d), // Last Power
+        Kilowatts(-55d), // PEl
+        Kilowatts(-55d), // PEl Solution
+      ), // increase, within load gradient
+      (
+        Kilowatts(-50d), // Last Power
+        Kilowatts(-41d), // PEl
+        Kilowatts(-41d), // PEl Solution
+      ), // decrease, within load gradient
+      (
+        Kilowatts(-30d), // Last Power
+        Kilowatts(-15d), // PEl
+        Kilowatts(-20.5d), // PEl Solution
+      ), // decrease, more than load gradient
+    )
+
+    testCases.foreach { case (lastPower, pEl, pElSol) =>
+      bmModel._lastPower = Some(lastPower)
+      val pElCalc = bmModel.applyLoadGradient(pEl)
+      pElCalc should approximate(pElSol)(power2Tolerance)
+    }
+  }
+
+  "calculate P correctly" in {
+
+    val testCases = Table(
+      ("time", "temp", "costControlled", "lastPower", "powerSol"),
+      (
+        "2019-01-05T05:15:00+01:00[Europe/Berlin]",
+        Celsius(10),
+        true,
+        Kilowatts(-40.0),
+        Kilowatts(-49.5),
+      ),
+      (
+        "2019-01-04T05:15:00+01:00[Europe/Berlin]",
+        Celsius(10),
+        true,
+        Kilowatts(-80.0),
+        Kilowatts(-89.5),
+      ),
+      (
+        "2019-01-04T05:15:00+01:00[Europe/Berlin]",
+        Celsius(-20),
+        true,
+        Kilowatts(-182.0),
+        Kilowatts(-190),
+      ),
+      (
+        "2019-01-04T05:15:00+01:00[Europe/Berlin]",
+        Celsius(-7),
+        true,
+        Kilowatts(-182.0),
+        Kilowatts(-190),
+      ),
+      (
+        "2019-01-04T05:15:00+01:00[Europe/Berlin]",
+        Celsius(-7),
+        false,
+        Kilowatts(-150.0),
+        Kilowatts(-152.16900643778735),
+      ),
+      (
+        "2019-07-07T10:15:00+02:00[Europe/Berlin]",
+        Celsius(19),
+        true,
+        Kilowatts(-10.0),
+        Kilowatts(-19.5),
+      ),
+      (
+        "2019-07-05T05:15:00+02:00[Europe/Berlin]",
+        Celsius(20),
+        true,
+        Kilowatts(-20.0),
+        Kilowatts(-29.5),
+      ),
+      (
+        "2019-07-06T10:15:00+02:00[Europe/Berlin]",
+        Celsius(20),
+        true,
+        Kilowatts(0.0),
+        Kilowatts(-9.5),
+      ),
+      (
+        "2019-07-05T05:15:00+02:00[Europe/Berlin]",
+        Celsius(22),
+        true,
+        Kilowatts(-22.0),
+        Kilowatts(-31.5),
```

Comment:
isnt this -12.5 kW in the groovy test?

Reply:
also other tests have a different outcome, is there a reason?

Reply:
Okay, now it works. I couldn't figure out why it calculated completely different values before, but now it's running. What do you think?

Reply:
The old test expected -12.5 kW as result, now we have -31.5 kW. There should be a reason or something is wrong. Let's search for it...

Reply:
Now it works :) Wrong table